pkg/deploy/stack_deployer.go

```diff
@@ -130,7 +130,7 @@ func NewTgGcFn(tgSynth *lattice.TargetGroupSynthesizer) TgGcCycleFn {
 }
 
 type TgGc struct {
-	lock    sync.Mutex
+	lock    sync.RWMutex
```

Comment:
The intent with this change is to many reconciliations to process in parallel, so long as target group GC isn't happening and vice versa. Previously, all reconciliations were serialized as a result of this lock. Since every `backendRef` gets a unique target group in Lattice, even when the `backendRefs` reference the same Kubernetes service, parallelized route reconciliation should be fine to run in parallel since they're not mutating any of the same resources.

As the logic grows in complexity, we may need to consider something like locks on individual target groups rather than a single "lock them all" lock.